src/Hpack/Config.hs

```diff
@@ -588,8 +589,8 @@ instance FromValue GitHub where
   fromValue v = do
     input <- fromValue v
     case map T.unpack $ T.splitOn "/" input of
-      [owner, repo, subdir] -> return $ GitHub owner repo (Just subdir)
-      [owner, repo] -> return $ GitHub owner repo Nothing
+      (owner: _) | owner `elem` ["http:", "https:"] -> fail $ "expected owner/repo or owner/repo/subdir, but encountered url instead: " ++ show input
```

Comment:
Had to add some hacky url detection because otherwise urls were being matched as
```
owner: "https:"
repo: ""
subdir: "github.com/sol/hpack/issues/365"
```

Not sure how to handle this robustly